astropy/coordinates/tests/test_geodetic_representations.py

```diff
@@ -159,4 +203,103 @@ def test_geodetic_subclass_missing_equatorial_radius():
         class MissingCustomGeodeticAttribute(BaseGeodeticRepresentation):
             _flattening = 0.075 * u.dimensionless_unscaled
 
-    assert "customgeodeticerror" not in REPRESENTATION_CLASSES
+    assert "missingcustomgeodeticattribute" not in REPRESENTATION_CLASSES
+
+
+def test_geodetic_subclass_bad_attributes():
+    msg = "Invalid argument 'foo' for '_positive_logitude'"
+    with pytest.raises(ValueError, match=msg):
+
+        class BadPositiveLongitudeCustomGeodetic(BaseGeodeticRepresentation):
+            _equatorial_radius = 3000.0 * u.km
+            _flattening = 0.075 * u.dimensionless_unscaled
+            _positive_longitude = "foo"
+
+    assert "badpositivelongitudecustomgeodetic" not in REPRESENTATION_CLASSES
+
+    msg = "Invalid argument 'foo' for '_wrap_angle'"
+    with pytest.raises(ValueError, match=msg):
+
+        class BadWrapAngleCustomGeodetic(BaseGeodeticRepresentation):
+            _equatorial_radius = 3000.0 * u.km
+            _flattening = 0.075 * u.dimensionless_unscaled
+            _wrap_angle = "foo"
+
+    assert "badwrapanglecustomgeodetic" not in REPRESENTATION_CLASSES
+
+
+def test_from_cartesian_positive_direction_180_360():
+    # Test array-valued input in the process.
+    initial_cartesian = CartesianRepresentation(
+        x=[1, 3000.0, 2600.0] * u.km,
+        y=[7000.0, 4.0, -5000.0] * u.km,
+        z=[5.0, 6000.0, 3000.0] * u.km,
+    )
+
+    assert_quantity_allclose(
+        (
+            IAUMARS2000GeodeticRepresentationEast180.from_representation(
+                initial_cartesian
+            ).lon.value
```

Comment:
Just get `.lon` - there should be no need to get `.value` and then multiple with `u.deg`.

Also, I think here and below I would just check the output directly (with numbers) rather than rely on math.